### PR TITLE
add saving and loading infairness component state to disk

### LINKED ISSUE #56: allow saving and loading inFairness models to disk

### PR DESCRIPTION
**Pull request by MayankAgarwal**
fixes #56 

This PR adds support for saving and loading trained distance metrics to disk. The format is the same as PyTorch modules, like:

```
dist = distances.SVDSensitiveSubspaceDistance()   # can be any metric
dist.fit(*args, **kwargs)

# saving state to disk
state = dist.state_dict()
torch.save(state, PATH)

# load state from disk
state = torch.load(PATH)
dist = distances.SVDSensitiveSubspaceDistance()
dist.load_state_dict(state)
```

Refer: https://pytorch.org/tutorials/beginner/saving_loading_models.html#saving-loading-model-for-inference

### CODE REVIEW

**Review by moonfolk — APPROVED** [review]
Can we also add a way to access logistic regression coefficients (including bias - it is often useful to check LR accuracy when learning the fair metric) in this PR or do you want to do a different PR for it? I am suggesting it here since we also want to save it along with sigma probably?
Another comment, which could be a separate PR, we want to provide additional input arguments when fitting LR such as optimizing for balanced accuracy via `class_weight='balanced'`.
